src/nexmo/__init__.py

```diff
@@ -624,69 +593,58 @@ def parse(self, host, response):
             # Strip off any encoding from the content-type header:
             content_mime = response.headers.get("content-type").split(";", 1)[0]
             if content_mime == "application/json":
-                return response.json()
+                #Check for exceptions before retrieve data
+                data = response.json()
+                if "messages" in data and self.__error_handler.validate_code(data["messages"][0]["status"]):
```

Comment:
this feels really off here - this parse method is supposed to be really low level IMO it should only parse the response - pull out the JSON or the other response content and return that up a level. Parse here is being commissioned to do some pretty API specific error handling - that handling should be done in the send_messages request.
> Note we have other APIs that have a similar schema of return a 2xx + error code to message that they've failed - I think it would be great if you have the error handling handled in the methods used for making those calls.

Reply:
This point still holds I think - this logic should be managed by the client functions of this. This will leave us flexible to add the same behavior for Numbers, NumbersInsights, Verify which have similar behavior.